src/hasSubObject.ts

```diff
@@ -19,10 +21,9 @@ import type { Simplify } from "./type-fest/simplify";
  * @category Guard
  */
 export function hasSubObject<T, S extends Partial<T>>(
-  data: T,
+  data: Merge<T, S> | T,
   subObject: S,
-): // TODO: want to use data is Merge<T, S> here, but it's a typescript error. fix after we allow @ts-expect-error in the build process
-data is Simplify<S & T>;
+): data is HasSubObjectBrand & Merge<T, S>;
```

Comment:
Reverted this to `Simplify<S & T>` in https://github.com/remeda/remeda/pull/686 because `Merge` is quite a complex type and I couldn't make `pipe` to work well in this case.  `Simplify` worked well in all cases.